From 82d7c5f03c493244b1f0f7472afbc7282a940b9d Mon Sep 17 00:00:00 2001
From: "Alexander E. Patrakov" <patrakov@gmail.com>
Date: Fri, 13 Dec 2013 17:47:38 +0600
Subject: [PATCH 2/4] pavucontrol: ellipsize labels to fix window resizing

ellipsize labels to make the window resizable even with long label text
and add tooltips to provide a way to read the full text

Originally from Sebastian Wick <sebastian@sebastianwick.net>
---
 src/mainwindow.cc     | 6 ++++++
 src/pavucontrol.glade | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/src/mainwindow.cc b/src/mainwindow.cc
index 8fa3949..5a42318 100644
--- a/src/mainwindow.cc
+++ b/src/mainwindow.cc
@@ -433,6 +433,7 @@ bool MainWindow::updateSink(const pa_sink_info &info) {
     w->boldNameLabel->set_text("");
     gchar *txt;
     w->nameLabel->set_markup(txt = g_markup_printf_escaped("%s", info.description));
+    w->nameLabel->set_tooltip_text(info.description);
     g_free(txt);
 
     icon = pa_proplist_gets(info.proplist, PA_PROP_DEVICE_ICON_NAME);
@@ -588,6 +589,7 @@ void MainWindow::updateSource(const pa_source_info &info) {
     w->boldNameLabel->set_text("");
     gchar *txt;
     w->nameLabel->set_markup(txt = g_markup_printf_escaped("%s", info.description));
+    w->nameLabel->set_tooltip_text(info.description);
     g_free(txt);
 
     icon = pa_proplist_gets(info.proplist, PA_PROP_DEVICE_ICON_NAME);
@@ -709,6 +711,8 @@ void MainWindow::updateSinkInput(const pa_sink_input_info &info) {
         w->nameLabel->set_label(info.name);
     }
 
+    w->nameLabel->set_tooltip_text(info.name);
+
     setIconFromProplist(w->iconImage, info.proplist, "audio-card");
 
     w->setVolume(info.volume);
@@ -761,6 +765,8 @@ void MainWindow::updateSourceOutput(const pa_source_output_info &info) {
         w->nameLabel->set_label(info.name);
     }
 
+    w->nameLabel->set_tooltip_text(info.name);
+
     setIconFromProplist(w->iconImage, info.proplist, "audio-input-microphone");
 
 #if HAVE_SOURCE_OUTPUT_VOLUMES
diff --git a/src/pavucontrol.glade b/src/pavucontrol.glade
index 4b3dd01..b0aa2a5 100644
--- a/src/pavucontrol.glade
+++ b/src/pavucontrol.glade
@@ -256,6 +256,7 @@
                             <property name="xalign">0</property>
                             <property name="label" translatable="yes">Device Title</property>
                             <property name="use_markup">True</property>
+                            <property name="ellipsize">end</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -1384,6 +1385,7 @@
                             <property name="xalign">0</property>
                             <property name="label" translatable="yes">Stream Title</property>
                             <property name="use_markup">True</property>
+                            <property name="ellipsize">end</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
-- 
2.1.0

